iamclient/iamclient.go

```diff
@@ -51,32 +47,56 @@ const (
 	iamReconnectInterval = 10 * time.Second
 )
 
+const iamCertType = "iam"
+
 /***********************************************************************************************************************
  * Types
  **********************************************************************************************************************/
 
 // Client IAM client instance.
 type Client struct {
 	sync.Mutex
+	*grpchelpers.IAMPublicServiceClient
 
-	sender Sender
+	config        *config.Config
+	sender        Sender
+	cryptocontext *cryptutils.CryptoContext
+	insecure      bool
 
 	nodeID   string
 	systemID string
 
-	publicConnection    *grpc.ClientConn
+	publicConnection    *grpchelpers.BlockingGRPCConn
 	protectedConnection *grpc.ClientConn
-	publicService       pb.IAMPublicServiceClient
+
 	identService        pb.IAMPublicIdentityServiceClient
 	certificateService  pb.IAMCertificateServiceClient
 	provisioningService pb.IAMProvisioningServiceClient
+	publicNodesService  pb.IAMPublicNodesServiceClient
+	nodesService        pb.IAMNodesServiceClient
+
+	nodeInfoSubs *nodeInfoChangeSub
+
+	tlsCertChan      <-chan *pb.CertInfo
+	closeChannel     chan struct{}
+	disableReconnect atomic.Bool
+	reconnectChannel chan struct{}
+
+	reconnectTimer *time.Timer
+}
 
-	closeChannel       chan struct{}
-	publicNodesService pb.IAMPublicNodesServiceClient
-	nodesService       pb.IAMNodesServiceClient
-	nodeInfoListeners  []chan cloudprotocol.NodeInfo
+// certSubscription generic subscription for IAM public messages.
+type iamSubscription[grpcStream *pb.IAMPublicService_SubscribeCertChangedClient |
```

Comment:
Where is `IAMPublicService_SubscribeCertChangedClient` used?

Reply:
fixed in aos_common